frontend/libs/studio-components-legacy/src/components/StudioCodelistEditor/StudioCodeListEditor.tsx

```diff
@@ -65,7 +67,19 @@ function StatefulCodeListEditor({
   onInvalid,
   textResources,
 }: StatefulCodeListEditorProps): ReactElement {
-  const [codeList, setCodeList] = usePropState<CodeList>(defaultCodeList);
+  const initialState: ReducerState = {
+    codeList: defaultCodeList,
+    textResources: textResources ?? [],
+  };
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    dispatch({
+      type: ReducerActionType.SetState,
+      codeList: defaultCodeList,
+      textResources: textResources,
+    });
+  }, [defaultCodeList, textResources]);
```

Comment:
Jeg tror vi bør være forsiktige med å oppdatere hele datastrukturen i én vending på denne måten. Hvis komponenten blir lastet på nytt etter at tekstressursene er oppdatert, men før kodelisten blir det, vil vi få en situasjon med usynkronisert data fordi denne funksjonen overskriver den nye kodelisten med den gamle. Har du vurdert å lage to separate `useEffect`er?